nixos/modules/services/video/epgstation/default.nix

```diff
@@ -0,0 +1,297 @@
+        example = "/run/keys/epgstation-password";
+        description = ''
+          A file containing the password corresponding to
+          <option>basicAuth.user</option>.
+        '';
+      };
+    };
+
+    mirakurunPath =  mkOption (let
+      # Should be kept in sync with services.mirakurun.serverSettings.path
+      sockPath = "/var/run/mirakurun/mirakurun.sock";
+    in {
+      type = types.str;
+      default = "http+unix://${replaceStrings ["/"] ["%2F"] sockPath}";
+      example = "http://localhost:40772";
+      description = "URL to connect to Mirakurun.";
+    });
+
+    database =  {
+      name = mkOption {
+        type = types.str;
+        default = "epgstation";
+        description = "Name of the database that holds EPGStation's data.";
+      };
+
+      password = mkOption {
+        type = with types; nullOr str;
+        default = "epgstation";
+        description = ''
+          Database password for EPGStation.
+
+          Warning: This is implemented to store the plain text in the nix store.
+        '';
+      };
+
+      passwordFile = mkOption {
+        type = with types; nullOr path;
+        default = pkgs.writeText "epgstation-db-password" cfg.database.password;
+        example = "/run/keys/epgstation-db-password";
+        description = ''
+          A file containing the password corresponding to the database
+          <option>database.name</option>.
+        '';
+      };
+
+      connectTimeout = mkOption {
+        type = types.ints.unsigned;
+        default = 1000;
+        description = "Connection timeout for the database in milliseconds";
+      };
+
+      connectionLimit = mkOption {
+        type = types.ints.unsigned;
+        default = 10;
+        description = "Maximum number of connections to the database.";
+      };
+    };
+
+    encode = mkOption {
+      type = with types; listOf attrs;
+      description = "Encoding presets for recorded videos.";
+      default = [
+        { name = "H264";
+          cmd = "${pkgs.epgstation}/libexec/enc.sh main";
+          suffix = ".mp4";
+          default = true; }
+        { name = "H264-sub";
+          cmd = "${pkgs.epgstation}/libexec/enc.sh sub";
+          suffix = "-sub.mp4"; }
+      ];
+    };
+
+    settings = mkOption {
+      type = settingsFmt.type;
+      default = {};
+      example = {
+        recPriority = 20;
+        conflictPriority = 10;
+      };
+      description = ''
+        Options which are addded to config.json.
+
+        Documentation:
+        <link xlink:href="https://github.com/l3tnun/EPGStation/blob/master/doc/conf-manual.md"/>
+      '';
+    };
+
+  };
+
+  config = mkIf cfg.enable {
+    environment.etc = {
+      "epgstation/operatorLogConfig.json".text = builtins.toJSON logConfig;
+      "epgstation/serviceLogConfig.json".text = builtins.toJSON logConfig;
+    };
+
+    networking.firewall = mkIf cfg.openFirewall {
+      # port (serverPort + 1) is used for websockets
```

Comment:
What exactly does this mean? Maybe it should also be documented in the `serverPort` option.

Reply:
Epgstation listens on a total of 2 ports, one for HTTP and one for websockets. It needs both for the webapp to function.

Reply:
Fixed description for `serverPorts`.